app/angular/src/client/preview/angular/ComponentClassFromStoryComponent.ts

```diff
@@ -0,0 +1,115 @@
+import {
+  AfterViewInit,
+  ChangeDetectorRef,
+  Component,
+  ElementRef,
+  Inject,
+  OnDestroy,
+  Type,
+  ViewChild,
+  ViewContainerRef,
+} from '@angular/core';
+import { Subscription, Subject } from 'rxjs';
+import { map, skip } from 'rxjs/operators';
+
+import { ICollection } from '../types';
+import { STORY_PROPS } from './app.token';
+import { RenderNgAppService } from './RenderNgAppService';
+
+const findComponentDecoratorMetadata = (component: any) => {
+  const decoratorKey = '__annotations__';
+  const decorators: any[] =
+    Reflect &&
+    Reflect.getOwnPropertyDescriptor &&
+    Reflect.getOwnPropertyDescriptor(component, decoratorKey)
+      ? Reflect.getOwnPropertyDescriptor(component, decoratorKey).value
+      : (component as any)[decoratorKey];
+
+  const ngComponentDecorator: Component | undefined = decorators.find(
+    (decorator) => decorator instanceof Component
+  );
+
+  return ngComponentDecorator;
+};
+
+const toInputsOutputs = (props: ICollection = {}) => {
+  return Object.entries(props).reduce(
+    (previousValue, [key, value]) => {
+      if (typeof value === 'function') {
+        return { ...previousValue, outputs: { ...previousValue.outputs, [key]: value } };
+      }
+
+      return { ...previousValue, inputs: { ...previousValue.inputs, [key]: value } };
+    },
+    { inputs: {}, outputs: {} } as { inputs: Record<string, any>; outputs: Record<string, any> }
+  );
+};
```

Comment:
`typeof value === 'function'` is not enough to determine an output.

The following story will incorrectly think the input is an output:
```ts
@Component({
  selector: 'story-example',
  template: `[{{ filter ? filter(list) : list }}]`
})
class StoryExample {
  @Input() filter: (list: string[]) => string[]

  list: string[] = [ 'a', 'b', 'c' ]
}

export const Example = () => ({
  props: {
    filter: (list: string[]) => list.filter(x => x !== 'b')
  },
  component: StoryExample
})
```

Reply:
if we have the right list of Inputs/Outputs from the component
we could make the distinction with it. 

And for all the props not found, it is that they are local to the component. We could also add them to the component in the ngAfterViewInit only once at the init.
This would keep the current behavior that allows to modify these properties. 🤷‍♂️ 

I'm wrong? if you're ok with it I can commit with this behavior.